uwuno/src/main/java/com/learning/uwuno/services/gameService.java

```diff
@@ -56,12 +58,70 @@ public player getPlayer(String uid, String pid) {
     }
 
     // PUTS
-    public room updateRoomName(String uid, String newName) {
+    public void updateRoomName(String uid, String newName) {
         if (newName.isBlank())
             throw new badRequest();
         room room = roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
         room.setRoomName(newName);
-        return room;
+    }
+
+    public void updateRoomStatus(String uid, String status) {
+        if (status.isBlank())
+            throw new badRequest();
+        room room = roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+        room.Status roomStatus = utils.stringToRoomState(status);
+
+        if (validStatusChange(room, roomStatus)) {
+            room.setRoomStatus(roomStatus);
+            setUpGameState(room, roomStatus);
+        } else {
+            throw new badRequest();
+        }
+    }
+
+    // TODO: Only has Lobby -> Start check. May need to add other states as necessary.
+    public boolean validStatusChange(room room, room.Status status) {
+        switch (status) {
+            case Lobby -> {
+                return true;
+            }
+            case Start -> {
+                return room.getPlayers().size() >= room.getMinPlayers() &&
+                        room.getPlayers().size() <= room.getMaxPlayers();
+            }
+            case End -> {
+                return false;
+            }
+        }
+        throw new badRequest();
```

Comment:
Is this option possible, I assume the the enum can only be one of the 3 above?

Reply:
Yes, it should be caught beforehand anyways. I can change it to return false.

Reply:
Maybe a comment saying will never get here and a internal server error instead?

Reply:
sure